### PR TITLE
add the verifySignature function to WalletClient.ts

### LINKED ISSUE #181: Add a "verifySignature" function
In the web3 folder we have functions to sign messages but we don't have any function to verify them.

verifySignature(message, signature, signerPubKey) -> return bool

### PR DESCRIPTION
**Pull request by Elli610**
closes #181

### CODE REVIEW

**Review by Ben-Rey — APPROVED** [review]

File: src/web3/WalletClient.ts [hunk 1]
```diff
@@ -377,6 +378,39 @@ export class WalletClient extends BaseClient implements IWalletClient {
...
+    data: string | Buffer,
+    signature: ISignature,
+    signerPubKey: string,
+  ): Promise<boolean> {
+    // setup the public key
+    const publicKeyBase58Decoded = WalletClient.getBytesPublicKey(signerPubKey);
+
+    // setup the message digest
+    const bytesCompact: Buffer = Buffer.from(data);
+    const messageHashDigest: Uint8Array = hashBlake3(bytesCompact);
```
Comment (gregLibert) [inline src/web3/WalletClient.ts:399]:
```suggestion
    const messageDigest: Uint8Array = hashBlake3(bytesCompact);
```

File: src/web3/WalletClient.ts [hunk 2]
```diff
@@ -377,6 +378,39 @@ export class WalletClient extends BaseClient implements IWalletClient {
...
+
+    // setup the signature
+    const hexSignature = base58Decode(signature.base58Encoded);
```
Comment (gregLibert) [inline src/web3/WalletClient.ts:402]:
```suggestion
    const signatureBytes = base58Decode(signature.base58Encoded);
```

Comment (gregLibert) [reply src/web3/WalletClient.ts:402]: Why not setting a type? this is not detected by the linter in the CI ?

File: src/web3/WalletClient.ts [hunk 3]
```diff
@@ -377,6 +378,39 @@ export class WalletClient extends BaseClient implements IWalletClient {
...
+
+    // verify signature
+    const isVerified = await ed.verify(
```
Comment (gregLibert) [inline src/web3/WalletClient.ts:405]: Why using a isVerified variable ? I think that ed.verify function name is sufficently explicite, no ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: src/web3/WalletClient.ts (hunk 2 above)
Comment (Elli610) [reply src/web3/WalletClient.ts:402]: looks like it is not detected.

File: src/web3/WalletClient.ts [hunk 4]
```diff
@@ -67,6 +67,7 @@ export class WalletClient extends BaseClient implements IWalletClient {
     this.sellRolls = this.sellRolls.bind(this);
     this.buyRolls = this.buyRolls.bind(this);
     this.getAccountBalance = this.getAccountBalance.bind(this);
+    this.verifySignature = this.verifySignature.bind(this);
```
Comment (gregLibert) [inline src/web3/WalletClient.ts:70]: Why do you need to bind this ?

**Review by gregLibert — COMMENTED** [review]
:question:

**Review by gregLibert — APPROVED** [review]

**Pull request merged**